### ISSUE TITLE
Allow MemoizedInstanceVariableName to support leading underscores

### ISSUE THREAD

**Issue by leklund** [rubocop-hq/rubocop#5843]
The MemoizedInstanceVariableName cop added in #5445 does not allow for instance variable names that have leading underscores -- `@_foo ||= get_foo`.

This is a common enough style pattern that the cop should have an enforced style option to require or optionally allow leading underscores.

--------

## Expected behavior

This should produce no offenses:

```
def my_method
  @_my_method ||= expensive_method_call
end
```
## Actual behavior

The MemoizedInstanceVariableName registers an offense.

Relevant PR comments: https://github.com/bbatsov/rubocop/pull/5445#issuecomment-375963377

**Comment by mikegee** [comment]
I think it is unfortunate that a class might have such complex instance variable usage to need a naming scheme at all, but obviously Rubocop can't ask for extensive refactoring. 🤷
